web/server/codechecker_server/api/mass_store_run.py

```diff
@@ -948,6 +949,53 @@ def get_missing_file_ids(report: Report) -> List[str]:
 
         return True
 
+    def __check_report_count(self, report_dir: Path):
+        """
+        This method checks the number of reports in the given report,
+        Raises exception if the number of reports is more than the
+        that is configured for the product.
+        """
+        # Read all plist files in report directory.
+        plist_files = list(report_dir.glob('**/*.plist'))
+        report_count = 0
+        for plist_file in plist_files:
+            # The check_name tag only occurs exactly once per report.
+            # Don't parse plist, just count the number of
+            # check_name occurances.
+            unique_tag = b'check_name'
+            with plist_file.open('rb') as f:
+                plist_file_content = f.read()
+                report_count += plist_file_content.count(unique_tag)
+
+        report_limit = 0
+        with DBSession(self.__config_database) as session:
+            product = session.query(Product).get(self.__product.id)
+            if product.report_limit:
+                report_limit = product.report_limit
+
+        if report_count > report_limit:
+            LOG.error("The number of reports in the given zip file is " +
+                      "larger than the allowed." +
+                      f"The limit is {report_limit}, you have: {report_count}!")
+            extra_info = [
+                "report_limit",
+                f"limit:{report_limit}"),
+                f"report_count:{report_count}"
+            ]
+            raise codechecker_api_shared.ttypes.RequestFailed(
+                codechecker_api_shared.ttypes.
+                ErrorCode.GENERAL,
+                "**Warning: Excessive Reports in This Report Folder** " +
+                "There are too many reports in this report " +
+                "folder, which has triggered a server " +
+                "protection measure. To proceed, please " +
```

Comment:
Holy shit this sounds apocalyptic :D
PROTECTIVE MEASURES TRIGGERED
ACTIVATE CONTINGENCY PLAN B007
MOVEMENT OF PERSONNEL IN OR OUT OF THE SERVER ROOM IS PROHIBITED

Reply:
Ehhh, you gotta have some fun. If we can't have even this much, we might as well be developing the Clang-Tidy!

Reply:
I will remove the server protection measure part from this too.